document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -249,6 +293,25 @@ public Document upsertAndReturn(Key key, Document document) throws IOException {
     return dbObjectToDocument(upsertResult);
   }
 
+  @SuppressWarnings("MismatchedQueryAndUpdateOfCollection")
+  private List<BasicDBObject> prepareForCreateOrReplace(final Key key, final Document document)
+      throws JsonProcessingException {
+    final long now = System.currentTimeMillis();
+
+    // Explicit createdTime field projection automatically unsets all the other existing fields
```

Comment:
Which other fields? Referring only to CREATED_TIME?

Reply:
All the other fields in the document are unset.
For example, if the existing document was
```json
{
   "foo": "bar",
   "createdTime": 1325374949,
   "updatedTime": 1325374949
}
```
projecting `createdTime` will automatically unset `foo` and `updatedTime`.